scalableminds/webknossos
# Disable editing of tracing dependent options in right tabs in read only mode

This PR disables all editing options of the right border tabs which are tracing dependent when the tracing is in read-only mode. This includes the skeleton tab, the comments tab, and the bounding box tab. Each disabled option gets a tooltip telling the user this option is disabled due to read-only mode. 

Note that I could not find any setting in the segments tab that seems to take effect in opening the same tracing in read-only mode. 


### URL of deployed dev instance (used for testing):
- https://disableeditinginreadonlymode.webknossos.xyz
- Due to making a dataset public, it might be better to test this locally.

### Steps to test:
- Create a hybrid tracing with some trees and group them. Create at least one bounding box, and comment a few skeleton nodes.
- Then make the dataset publicly available (or choose an already public available dataset). Next also make the tracing public and open it via the sharing link in a private / incognito tab.
- Try to edit the skeletons, commends, and bounding boxes in the read-only version of the tracing in the private tab. It should not be possible except for a few settings like visibility and sorting.

### Issues:
- fixes #6136

------
(Please delete unneeded items, merge only when none are left open)
